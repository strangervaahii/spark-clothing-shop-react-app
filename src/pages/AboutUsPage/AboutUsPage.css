.hero-title {
  font-weight: 700;
}
.hero-subtitle {
  font-weight: 500;
}
.banner {
  width: 100%;
  height: 300px;
}
.text-shadow {
  text-shadow: 1px 1px 3px rgba(0, 0, 0, 0.3);
}
