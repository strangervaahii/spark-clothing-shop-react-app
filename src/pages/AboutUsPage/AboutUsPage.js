// Importing necessary dependencies from React and react-router-dom libraries
import React from 'react';
import { NavLink, Outlet } from 'react-router-dom';
// Importing the HelmetSetup component and AboutUsPage's specific CSS styles
import HelmetSetup from '../../components/HelmetSetup/HelmetSetup';
import './AboutUsPage.css';

// Defining the functional component, AboutUsPage
const AboutUsPage = () => {
  return (
    <>
      {/* Setting up the page title with HelmetSetup */}
      <HelmetSetup title='AboutUsPage' />
      {/* Creating a banner section */}
      <div className='banner bg-primary'>
        <div className='container text-white text-center my-0 py-5'>
          <p className='fs-1 hero-title mb-3'>About Us</p>
          <p className='fs-4 hero-subtitle mb-3'>We have the capabilities and experience to deliver the products you need to move forward.</p>
          {/* Adding a button with a NavLink to the 'history' route, styled with a light background and black text */}
          <button type='button' className='btn btn-light shadow-none px-5'>
            <NavLink to='history' className='text-black link-underline-light fs-4'>History</NavLink>
          </button>
        </div>
      </div>
      {/* Creating a section with four cards displaying the company's mission, vision, values, and contact information */}
      <section>
        <div className='container'>
          <div className='row'>
            <div className='col-md-3 mt-2 border-0'>
              <div className='card h-100'>
                <div className='card-body'>
                  <img src='./assets/images/mission.png' className='card-img-top' alt='mission' />
                  <i className='fa-solid fa-bullseye-arrow'></i>
                  {/* Adding a title and text to the card body */}
                  <h3 className='card-title text-center font-weight-bold'>Mission</h3>
                  <p className='card-text'>Empowering customers worldwide with easy access to high-quality products
                    at competitive prices, delivered with exceptional customer service.</p>
                </div>
              </div>
            </div>
            {/* Repeating the card structure for vision, values, and contact information */}
            <div className='col-md-3 mt-2 border-0'>
              <div className='card h-100'>
                <div className='card-body'>
                  <img src='./assets/images/vision.jpg' className='card-img-top my-1' alt='vision' />
                  {/* Adding a title and text to the card body */}
                  <h3 className='card-title text-center font-weight-bold'>Vision</h3>
                  <p className='card-text'>To become the most trusted and preferred online shopping destination for customers,
                    offering a seamless and personalized shopping experience that exceeds expectations,
                    and fostering long-term relationships based on trust and loyalty.</p>
                </div>
              </div>
            </div>
            {/* Repeating the card structure for vision, values, and contact information */}
            <div className='col-md-3 mt-2 border-0'>
              <div className='card h-100'>
                <div className='card-body'>
                  <img src='./assets/images/values.png' className='card-img-top my-1' alt='values' />
                  {/* Adding a title and text to the card body */}
                  <h3 className='card-title text-center font-weight-bold'>Values</h3>
                  <p className='card-text'>At our core, we value transparency, integrity, and innovation, and are committed to
                    providing a safe and secure platform that prioritizes the needs and satisfaction of our customers,
                    employees, and partners.</p>
                </div>
              </div>
            </div>
            {/* Repeating the card structure for vision, values, and contact information */}
            <div className='col-md-3 mt-2 border-0'>
              <div className='card h-100'>
                <div className='card-body'>
                  <img src='./assets/images/contact.png' className='card-img-top my-1' alt='contact' />
                  {/* Adding a title and text to the card body */}
                  <h3 className='card-title text-center font-weight-bold'>Contact</h3>
                  <p className='card-text'>Need assistance? Our customer support team is available 24/7 to
                    assist with any questions or concerns. Contact us via phone, email, or live
                    chat to get the help you need, when you need it.</p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <Outlet />
    </>
  );
};
// Export the AboutUsPage component for use in other files
export default AboutUsPage;
