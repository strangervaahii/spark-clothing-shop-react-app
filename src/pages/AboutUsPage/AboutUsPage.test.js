// This code imports the necessary dependencies for testing the AboutUsPage
import React from 'react';
import { render, screen } from '@testing-library/react';
import AboutUsPage from './AboutUsPage';
import { HashRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

// This code begins the test suite for the AboutUsPage component.
describe('AboutUsPage component', () => {
  // This test checks if the Mission card is rendered.
  it('renders mission card', () => {
    render(<HashRouter><AboutUsPage /></HashRouter>);
    const missionCardTitle = screen.getByText('Mission');
    expect(missionCardTitle).toBeInTheDocument();
  });

  // This test checks if the Vision card is rendered.
  it('renders vision card', () => {
    render(<HashRouter><AboutUsPage /></HashRouter>);
    const visionCardTitle = screen.getByText('Vision');
    expect(visionCardTitle).toBeInTheDocument();
  });

  // This test checks if the Values card is rendered.
  it('renders values card', () => {
    render(<HashRouter><AboutUsPage /></HashRouter>);
    const valuesCardTitle = screen.getByText('Values');
    expect(valuesCardTitle).toBeInTheDocument();
  });

  // This test checks if the Contact card is rendered.
  it('renders contact card', () => {
    render(<HashRouter><AboutUsPage /></HashRouter>);
    const contactCardTitle = screen.getByText('Contact');
    expect(contactCardTitle).toBeInTheDocument();
  });

  // This test checks if the NavLink component has the correct href attribute.
  it('should render a NavLink component with the correct href attribute', () => {
    const { getByText } = render(<HashRouter><AboutUsPage /></HashRouter>);
    const historyLink = getByText('History');
    expect(historyLink).toHaveAttribute('href', '#/history');
  });

  // This test checks if the AboutUsPage component has the correct snapshot.
  it('has right snapshot with all requirements completed', () => {
    const snapshotInJson = renderer.create(<HashRouter><AboutUsPage /></HashRouter>).toJSON();
    // let's assert with toMatchSnapshot
    expect(snapshotInJson).toMatchSnapshot();
  })
});
