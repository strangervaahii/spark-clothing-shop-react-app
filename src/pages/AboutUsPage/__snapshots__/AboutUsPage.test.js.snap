// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUsPage component has right snapshot with all requirements completed 1`] = `
Array [
  <div
    className="banner bg-primary"
  >
    <div
      className="container text-white text-center my-0 py-5"
    >
      <p
        className="fs-1 hero-title mb-3"
      >
        About Us
      </p>
      <p
        className="fs-4 hero-subtitle mb-3"
      >
        We have the capabilities and experience to deliver the products you need to move forward.
      </p>
      <button
        className="btn btn-light shadow-none px-5"
        type="button"
      >
        <a
          className="text-black link-underline-light fs-4"
          href="#/history"
          onClick={[Function]}
        >
          History
        </a>
      </button>
    </div>
  </div>,
  <section>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-3 mt-2 border-0"
        >
          <div
            className="card h-100"
          >
            <div
              className="card-body"
            >
              <img
                alt="mission"
                className="card-img-top"
                src="./assets/images/mission.png"
              />
              <i
                className="fa-solid fa-bullseye-arrow"
              />
              <h3
                className="card-title text-center font-weight-bold"
              >
                Mission
              </h3>
              <p
                className="card-text"
              >
                Empowering customers worldwide with easy access to high-quality products at competitive prices, delivered with exceptional customer service.
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-md-3 mt-2 border-0"
        >
          <div
            className="card h-100"
          >
            <div
              className="card-body"
            >
              <img
                alt="vision"
                className="card-img-top my-1"
                src="./assets/images/vision.jpg"
              />
              <h3
                className="card-title text-center font-weight-bold"
              >
                Vision
              </h3>
              <p
                className="card-text"
              >
                To become the most trusted and preferred online shopping destination for customers, offering a seamless and personalized shopping experience that exceeds expectations, and fostering long-term relationships based on trust and loyalty.
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-md-3 mt-2 border-0"
        >
          <div
            className="card h-100"
          >
            <div
              className="card-body"
            >
              <img
                alt="values"
                className="card-img-top my-1"
                src="./assets/images/values.png"
              />
              <h3
                className="card-title text-center font-weight-bold"
              >
                Values
              </h3>
              <p
                className="card-text"
              >
                At our core, we value transparency, integrity, and innovation, and are committed to providing a safe and secure platform that prioritizes the needs and satisfaction of our customers, employees, and partners.
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-md-3 mt-2 border-0"
        >
          <div
            className="card h-100"
          >
            <div
              className="card-body"
            >
              <img
                alt="contact"
                className="card-img-top my-1"
                src="./assets/images/contact.png"
              />
              <h3
                className="card-title text-center font-weight-bold"
              >
                Contact
              </h3>
              <p
                className="card-text"
              >
                Need assistance? Our customer support team is available 24/7 to assist with any questions or concerns. Contact us via phone, email, or live chat to get the help you need, when you need it.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
